src/app/modules/submission/encrypt-submission/encrypt-submission.service.ts

```diff
@@ -70,11 +71,15 @@ export const transformAttachmentMetaStream = ({
     objectMode: true,
     transform: (data: SubmissionCursorData, _encoding, callback) => {
       const unprocessedMetadata = data.attachmentMetadata ?? {}
+
+      // Create new pure object to transform attachment data.
+      const transformedData: SubmissionCursorData = cloneDeep(data)
```

Comment:
why do we need to do this?

Reply:
We don't, it's just a matter of preference, trying to prevent as much side effects in code as possible. Should I change it back?

Reply:
yeah please do. stream transformations are inherently mutable, and object cloning is expensive. This will also reduce the likelihood of memory/heap thrashing from allocation/deallocating so many temporary objects.

Reply:
Good point, thanks for catching this! Reverted to mutable implementation in 42d00aa. Feel free to merge in if there are no other issues.